docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,819 @@
+  switch publicKeyDescriptor {
+  case "xpub", "tpub":
+    // Noop, the address is already correct
+  case "ypub", "upub":
+    // p2wpkh-in-p2sh, constructed as per https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#p2wpkh-nested-in-bip16-p2sh .
+    scriptSig = append([]byte{0x00, 0x14}, requestedAddress.Hash160()[:]...)
+    finalAddress = btcutil.NewAddressWitnessScriptHash(
+      btcutil.Hash160(scriptSig),
+      chainParams,
+    )
+  case "zpub", "vpub":
+    // p2wpkh
+    finalAddress, err = btcutil.NewAddressWitnessPubKeyHash(
+      requestedAddress.Hash160()[:],
+      chainParams,
+    )
+  }
+  if (err != nil) {
+    return fmt.Errorf(
+      "failed to derive final address format from extended key: [%s]",
+      err,
+    )
+  }
+
+  return finalAddress.EncodeAddress()
+}
+```
+
+[[transaction-construction]]
+==== Construction of Unsigned Transaction
+
+The unsigned transaction can be constructed from information about the
+previous transaction, a fee per vbyte, a set of recipient addresses, and a
+dummy signature and public key for the funding UTXO. This process results in
+an almost-complete transaction with the associated final fee. The sample
+function below shows the various inputs as well as how they can be combined
+to produce the transaction
+
+```go
+// import "github.com/btcsuite/btcd/wire"
+// import "github.com/btcsuite/btcd/txscript"
+
+func ConstructUnsignedTransaction(
+	previousOutputTransactionHashHex string, // <1>
+	previousOutputIndex uint32,
+	previousOutputValue int64, // <2>
+	feePerVbyte int64, // <3>
+	recipientAddresses []string,
+	chainParams *chaincfg.Params, // <4>
+) (*wire.MsgTx, error) {
+    // If the previous output transaction hash is passed as a []byte, can use
+    // chainhash.NewHash.
+	previousOutputTransactionHash, err := chainhash.NewHashFromStr(
+		previousOutputTransactionHashHex,
+	)
+	if err != nil {
+		return "", fmt.Errorf(
+			"error extracting outpoint transaction hash: [%s]",
+			previousOutputTransactionHash,
+		)
+	}
+
+	// The witness signature field is the DER signature followed by the hash type.
+  // We write a dummy signature with 73 0 bytes. DER signatures vary in encoding
+  // between 71, 72, and 73 bytes, so we choose the longest for fee purposes.
+	dummySignatureForWitness := []byte{
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0,
+    0, // one more dummy byte for the SigHashType
+	}
+  // The compressed public key requires 33 bytes.
+  dummyCompressedPublicKeyForWitness := []byte{
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0,
+  }
+
+	tx := wire.NewMsgTx(wire.TxVersion)
+	txIn := wire.NewTxIn(
+		wire.NewOutPoint(previousOutputTransactionHash, previousOutputIndex),
```

Comment:
I'm getting an error saying:
```
(type *"github.com/decred/dcrd/chaincfg/chainhash".Hash) as type *"github.com/btcsuite/btcd/chaincfg/chainhash".Hash in argument to wire.NewOutPoint
```

Are we able to typecast  `dcrd/chaincfg/chainhash.Hash` into `btcd/chaincfg/chainhash.Hash`? (it looks like they're both just `[]byte`)?

Reply:
I think this is actually because of some import nonsense that I'm working through - false alarm